s3/retryer.go

```diff
@@ -0,0 +1,48 @@
+package s3
+
+import (
+	"github.com/aws/aws-sdk-go/aws/client"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"time"
+)
+
+var defaultPermanentErrorCodes = map[int]struct{} {
+	404 : {},
+	418 : {}, // always don't retry if aws is teapot
```

Comment:
Funny error code)

Reply:
Also, 4xx means that our request is incorrect, so I think we shouldn't retry all 4xx errors